nc/nc.go

```diff
@@ -474,3 +476,36 @@ func (c *Client) Stop() error {
 
 	return nil
 }
+
+// Revoke revokes the NC communication token
+func (c *Client) Revoke(purgeSession bool) bool {
+	c.startMu.Lock()
+	defer c.startMu.Unlock()
+
+	if c.started {
+		log.Println(logPrefix, "attempt to revoke token for running client")
+		return false
+	}
+
+	var cfg config.Config
+	if err := c.credsFetcher.cm.Load(&cfg); err != nil {
```

Comment:
This entire block(line 490-510) should be encapsulated in `CredsFetcher`.

Reply:
Done.